geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/LiveFeatureCacheBenchmarkTest.scala

```diff
@@ -0,0 +1,395 @@
+    "What = 5",
+    "Who = 'Fred'",
+    "What = 6",
+    "Who = 'Goliath'",
+    "What = 7",
+    "Who = 'Harry'",
+    "What = 8"))
+
+  // Easier filters
+  val geoCD = ff.and(cd, bbox2)
+  val geoAB = ff.and(ab, bbox2)
+
+  val abcd = ff.or(ab, cd)
+
+  val abcdWhere = ff.and(abcd, where)
+  val abcdWhere2 = ff.and(abcd, where2)
+
+  val badOr3 = ff.or(geoCD, geoAB)
+  val niceAnd3 = ff.and(abcd, bbox2)
+  val jor = ff.or(abcdWhere, abcdWhere2)
+
+  // One geom
+  val whereAB = ff.and(where, ab)
+  val whereCD = ff.and(where, cd)
+
+  val bad1 = ff.or(whereAB, whereCD)
+  val nice1 = ff.and(where, abcd)
+
+  val filters = Seq(ab, cd, w14, where, justified, justifiedAB, justifiedCD, just, justBBOX, justBBOX2, bbox2, overlapOR1, overlapOR2, overlapORpathological)
+
+  val nFeats = 100000
+  val feats = (0 until nFeats).map(buildFeature)
+  val featsUpdate = (0 until nFeats).map(buildFeature)
+
+  // load different LiveFeatureCache implementations
+  implicit val ticker = Ticker.systemTicker()
+  val lfc = new LiveFeatureCacheGuava(sft, None)
+  //val h2  = new LiveFeatureCacheH2(sft)
+  val cq = new LiveFeatureCacheCQEngine(sft, None)
+
+  val sfv = new SimplifyingFilterVisitor
+
+  def benchmark(f: Filter) {
+    println("Running f")
+    val (regularCount, t1) = time(lfc.getReaderForFilter(f).getIterator.size)
+
+    println("\n\nRunning filter in CNF")
+    val cnf = rewriteFilterInCNF(f)
+    val (cnfCount, tcnf) = time(lfc.getReaderForFilter(cnf.accept(sfv, null).asInstanceOf[Filter]).getIterator.size)
+
+    println("\n\nRunning filter in DNF")
+    val dnf = rewriteFilterInDNF(f)
+    val (dnfCount, tdnf) = time(lfc.getReaderForFilter(dnf.accept(sfv, null).asInstanceOf[Filter]).getIterator.size)
+
+    println("\n\nRunning filter in 'unoptimized'")
+    //val (simpleCount, t2) = time(lfc.getReaderForFilter(f.accept(sfv, null).asInstanceOf[Filter]).getIterator.size)
+    val (unoptimizedCount, t3) = time(lfc.unoptimized(f).getIterator.size)
+
+    println(s"\nFilter: $f")
+    if (regularCount == cnfCount && regularCount == dnfCount && regularCount == unoptimizedCount) {
+      if (t3 < t1) {
+        println("'Unoptimized' was quicker than regular")
+      }
+      if (tcnf < t1) {
+        println("'CNF' was quicker than regular")
+      }
+      if (tdnf < t1) {
+        println("'DNF' was quicker than regular")
+      }
+      println(s"All filters returned $regularCount")
+
+    } else {
+      println(s"MISMATCHED Counts: Regular: $regularCount CNF: $cnfCount DNF: $dnfCount  Unoptimized: $unoptimizedCount")
+    }
+    println(s"Timings: regular: $t1 CNF: $tcnf DNF: $tdnf unoptimized: $t3\n")
+  }
+
+  def toC(f: Filter) = {
+    f match {
+      case or: Or => or.getChildren.toIndexedSeq
+      case _ => Seq(f)
+    }
+  }
+
+  // f is bigger than g
+  def contains(f: Filter, g: Filter): Boolean = {
+    val fc = toC(f)
+    val gc = toC(g)
+    fc.contains(gc)
+  }
+
+
+  def printBoolean(f: Filter): String = {
```

Comment:
Removed all of the code not used in the benchmark test at the bottom of this class.